### PR TITLE
ascent: add v0.9.3

### PR DESCRIPTION
**Pull request by white238**
This adds the new Ascent version.

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed the commit, rest LGTM.

**Comment by tldahlgren** [comment]
@cyrush @scheibelp Any objections to merging this PR?

**Comment by cyrush** [comment]
We also new VTK-m 2.1 w/ a patch.  And newer RAJA (which may be happening just b/c newer RAJA is a default in spack?)

**Comment by cyrush** [comment]
Maybe same for vtk-m?

Is there a way I can tell what concrete specs are being tested?

**Comment by scheibelp** [comment]
> Is there a way I can tell what concrete specs are being tested?

If you know which pipelines build the package being updated, you can check those (via the details of `ci/gitlab-ci` check). In this case it looks like the `radiuss-build` pipeline builds `ascent` and the run for this PR built `0.9.3`: https://gitlab.spack.io/spack/spack/-/pipelines/726683. We don't appear to provide easy access to details beyond version/compiler (e.g. variants used) - at least via those links.

**Review by scheibelp — DISMISSED** [review]

File: var/spack/repos/builtin/packages/ascent/package.py [hunk 1]
```diff
@@ -189,6 +193,7 @@ class Ascent(CMakePackage, CudaPackage):
     depends_on("raja~openmp", when="+raja ~openmp")
     depends_on("umpire", when="+umpire")
     depends_on("umpire@:2023.06.0", when="@:0.9.2 +umpire")
+    depends_on("umpire@2024.02:", when="@0.9.3: +umpire")
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/ascent/package.py]: It sounds like @cyrush is saying a similar constraint would make sense for `raja`. @white238 would you prefer to do that here? Based on checking the pipeline it looks like we got build coverage over your new version, but the constraint could help avoid problems in other contexts.

**Comment by cyrush** [comment]
These are the preferred major deps:

conduit@0.9.1
vtk-m@2.1.0 (requires [patch](https://github.com/Alpine-DAV/ascent/blob/0aef6cffd522be7419651e6adf586f9a553297d0/scripts/build_ascent/2024_05_03_vtkm-mr3215-ext-geom-fix.patch) )
raja@2024.02.1
umpire@2024.02.1
camp@2024.02.1
kokkos@3.7.02 (for rocm)

from: https://github.com/Alpine-DAV/ascent/blob/develop/CHANGELOG.md

I am fine with pinning them down exactly, but not sure if that is the current spack philosophy (pin in the package vs else where?) 

At a minimum, (except for conduit) -- things won't work with older versions than these deps. We could pin forward.

However future versions make (will?) change APIs.

**Comment by tldahlgren** [comment]
> > Is there a way I can tell what concrete specs are being tested?
> 
> If you know which pipelines build the package being updated, you can check those (via the details of `ci/gitlab-ci` check). In this case it looks like the `radiuss-build` pipeline builds `ascent` and the run for this PR built `0.9.3`: https://gitlab.spack.io/spack/spack/-/pipelines/726683. We don't appear to provide easy access to details beyond version/compiler (e.g. variants used) - at least via those links.

@cyrush When you say tested, do you mean built (as @scheibelp notes)? Spack only re-builds packages that it determines are needed to update the build cache.

Or are you referring to running package tests?  If that is the case, spack only runs the stand-alone tests for packages it rebuilds.  The easiest way to see test results for a given PR is to find and follow the pipeline link to get to the test outputs, which are currently being saved as artifacts.  In the above pipeline, you can see the results of the stand-alone test by  browsing (or downloading) the job artifacts:  `jobs_scratch_dir` -> `tests`.  The stand-alone test output is in the file whose name ends `test-out.txt`.    (The other way is to try to find the outputs in CDash at https://cdash.spack.io/index.php?project=Spack%20Testing.  It's not as easy to tie them to a specific PR though.)

**Comment by white238** [comment]
I'll push the constraints and patch that @cyrush listed tomorrow. Please don't merge this until I do.

**Comment by cyrush** [comment]
@tldahlgren  sorry for not being clear, I was wondering what build specs the CI tests are exercising. 
Things have grown pretty complex and broad, I haven't been able to keep up with understanding how to digest the firehose of CI dashboards.

**Comment by white238** [comment]
@spackbot fix style

**Comment by white238** [comment]
@adrienbernede This is the PR I was talking about needing `raja@2024.02.01` plus umpire/camp.

**Comment by tldahlgren** [comment]
> @tldahlgren sorry for not being clear, I was wondering what build specs the CI tests are exercising. Things have grown pretty complex and broad, I haven't been able to keep up with understanding how to digest the firehose of CI dashboards.

For `ascent` or all packages?  If the former, a quick check shows Spack CI is building default `ascent` plus `ecp-data-vis-sdk` with `~openmp` or `+openmp` but the package has no stand-alone tests to exercise.  The CDash build and test results for the different stacks are posted at https://cdash.spack.io/index.php?project=Spack%20Testing.  (Note that only packages that need to be re-built for the build cache are tested.)

**Comment by cyrush** [comment]
Thanks @tldahlgren.

After hunting around -  I see ascent 0.8.0 is being built. That version is from 2022, so I am not sure spack CI is exercising new versions and their related constraints.  I think those of those new cases are tricky - which is we are are pinned back in time.

I recall I added tests to the conduit pkg and then wanted to understand better process for things like discovering the  location of  cmake in the test env before moving forward with Ascent spack tests.  We have tests in Ascent that look just like the standalone conduit tests.

**Comment by adrienbernede** [comment]
@white238 I think this https://github.com/spack/spack/pull/44699 is what you were waiting for.

**Comment by white238** [comment]
This PR is ready to go now once it passes tests.

**Review by scheibelp — APPROVED** [review]

**Pull request merged**